src/components/Task/Task.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import PropType from 'prop-types';
+import Button from '../Button/Button';
+import classes from './Task.module.css';
+import Modal from '../Modals/Modal';
+import { TASK_MODAL, DELETE_TASK, openDeleteTaskModal, deleteTask } from '../../utilities/actionCreators';
+
+export default class Task extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedID: '',
+      isOpen: false,
+      selectedModal: '',
+    };
+    this.dispatch = this.dispatch.bind(this);
+  }
+
+  componentDidMount() {
+    const {
+      taskData: { id },
+    } = this.props;
+    this.setState((prevState) => ({
+      ...prevState,
+      selectedID: id,
+    }));
+  }
+
+  dispatch(action) {
+    const { dispatch } = this.props;
+    const { selectedID } = this.state;
+    switch (action.type) {
+      case TASK_MODAL:
+        this.setState((prevState) => ({
+          ...prevState,
+          isOpen: !prevState.isOpen,
+          selectedModal: action.modalType,
```

Comment:
You have the same code in several places, think how to avoid repetitive code

Reply:
Good advice, but difficult for solve! Right? 
```suggestion
switch (action.type) {
    case USER_MODAL:
    case TASK_MODAL:
      state = {
        ...prevState,
        isOpen: !prevState.isOpen,
        selectedModal: action.modalType,
      };
      return state;
    default:
      return prevState;
 }
```